### PR TITLE
AutomationWatchService uses relative path.

### PR DESCRIPTION
**Pull request by svilenvul**
See https://github.com/eclipse/smarthome/pull/2557#issuecomment-263563044

Adapting the code of AutomationWatchService to use relative path.
This change was necessary after #2432 was merged.

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/AutomationWatchService.java [hunk 1]
```diff
@@ -49,8 +51,9 @@ protected String getSourcePath() {
     }
 
     @Override
-    protected AbstractWatchQueueReader buildWatchQueueReader(WatchService watchService, Path toWatch) {
-        return new WatchQueueReader(watchService, toWatch, provider);
+    protected AbstractWatchQueueReader buildWatchQueueReader(WatchService watchService, Path toWatch,
+            Map<WatchKey, Path> registredWatchKeys) {
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/AutomationWatchService.java]: typo: registeredWatchKeys

File: bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/AutomationWatchService.java [hunk 2]
```diff
@@ -59,26 +62,29 @@ protected boolean watchSubDirectories() {
...
     }
 
     private static class WatchQueueReader extends AbstractWatchQueueReader {
 
         private AbstractFileProvider provider;
 
-        WatchQueueReader(WatchService watchService, Path dirToWatch, AbstractFileProvider provider) {
-            super(watchService, dirToWatch);
+        WatchQueueReader(WatchService watchService, Path dirToWatch, AbstractFileProvider provider,
+                Map<WatchKey, Path> registredWatchKeys) {
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/AutomationWatchService.java]: typo: registeredWatchKeys

**Comment by kaikreuzer** [comment]
> This change was necessary after #2342 was merged.

Is this the correct issue reference...?

**Comment by svilenvul** [comment]
>Is this the correct issue reference...?

Thanks, I've corrected it.

**Pull request merged**